ts/main.ts

```diff
@@ -376,3 +376,34 @@ function clearSelectedSqResetModal(sq: HTMLDivElement): void {
   $input.value = '';
   $modal.close();
 }
+
+const $gameSquares = document.querySelectorAll(
+  '.game-square',
+) as NodeListOf<HTMLDivElement>;
+if (!$gameSquares) throw new Error('.game-square query failed!');
+const $clearButton = document.querySelector('.clear-button');
+if (!$clearButton) throw new Error('.clear button query failed!');
+
+$clearButton.addEventListener('click', function () {
+  clearGrid();
+});
+
+function clearGrid(): void {
+  guesses = 0;
+  numCorrect = 0;
+  completedSquares = 0;
+  setStatsText();
+  $statsHeading.textContent = 'Stats';
+
+  for (let i = 0; i < $gameSquares.length; i++) {
+    console.log($gameSquares[i]);
+    $gameSquares[i].classList.remove(
+      'bg-green-500',
+      'bg-red-500',
+      'wrong',
+      'correct',
+    );
+    $gameSquares[i].classList.add('hover:bg-yellow-100');
+    $gameSquares[i].innerHTML = '';
+  }
```

Comment:
Why are you setting the `innerHTML` to an empty string? Doesn't this remove your elements? The other thing is that you don't want to use `innerHTML` because it could potentially run into security issues. A use could inject a string with a script tag and the browser would attempt to run the code in that script tag.

Reply:
the game-square divs are empty be default, once the user submits a correct answer a div image container and a nested img element are rendered and appended into the game-square divs. I use innerHTML = '' because it removes the image container and the img that were appended inside and resets the game-square divs back to how they were originally which is empty. Should I do something other than innerHTML?